py_cui/__init__.py

```diff
@@ -1098,34 +1115,37 @@ def _cycle_widgets(self, reverse=False):
         """
 
         num_widgets = len(self.get_widgets().keys())
-        current_widget_num = self._selected_widget
-
-        if not reverse:
-            next_widget_num = current_widget_num + 1
-            if self.get_widgets()[next_widget_num] is None:
-                if next_widget_num == num_widgets:
-                    next_widget_num = 0
-                next_widget_num = next_widget_num + 1
-            cycle_key = self._forward_cycle_key
-        else:
-            next_widget_num = current_widget_num - 1
-            if self.get_widgets()[next_widget_num] is None:
-                if next_widget_num < 0:
-                    next_widget_num = num_widgets - 1
-                next_widget_num = next_widget_num + 1
-            cycle_key = self._reverse_cycle_key
-
-        current_widget_id = current_widget_num
-        next_widget_id = next_widget_num
-        
-        if self._in_focused_mode and cycle_key in self.get_widgets()[current_widget_id]._key_commands.keys():
-            # In the event that we are focusing on a widget with that key defined, we do not cycle.
-            pass
-        else:
-            self.move_focus(self.get_widgets()[next_widget_id], auto_press_buttons=False)
+        current_widget_num: Optional[int] = self._selected_widget
+
+        if current_widget_num is not None:
+            if not reverse:
+                next_widget_num = current_widget_num + 1
+                if self.get_widgets()[next_widget_num] is None:
+                    if next_widget_num == num_widgets:
+                        next_widget_num = 0
+                    next_widget_num = next_widget_num + 1
+                cycle_key = self._forward_cycle_key
+            else:
+                next_widget_num = current_widget_num - 1
+                if self.get_widgets()[next_widget_num] is None:
+                    if next_widget_num < 0:
+                        next_widget_num = num_widgets - 1
+                    next_widget_num = next_widget_num + 1
+                cycle_key = self._reverse_cycle_key
+
+            current_widget_id: int = current_widget_num
+            next_widget_id: int = next_widget_num
+        current_widget = self.get_widgets()[current_widget_id] 
+        next_widget = self.get_widgets()[next_widget_id] 
+        if current_widget and next_widget is not None: #pls check again
+            if self._in_focused_mode and cycle_key in current_widget._key_commands.keys():
+                # In the event that we are focusing on a widget with that key defined, we do not cycle.
+                pass
+            else:
+                self.move_focus(next_widget, auto_press_buttons=False)
```

Comment:
Are these changes meant to fix something else?

Reply:
I have just included checks for None, to respect mypy.